libraries/lib-cloud-audiocom/OAuthService.cpp

```diff
@@ -0,0 +1,370 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  UploadService.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#include "OAuthService.h"
+
+#include <cassert>
+#include <cctype>
+
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+
+#include "CodeConversions.h"
+#include "Prefs.h"
+
+#include "IResponse.h"
+#include "NetworkManager.h"
+#include "Request.h"
+
+#include "ServiceConfig.h"
+
+#include "UrlDecode.h"
+
+#include "BasicUI.h"
+
+namespace cloud::audiocom
+{
+namespace
+{
+
+StringSetting refreshToken { L"/cloud/audiocom/refreshToken", "" };
```

Comment:
If it is stipulated in the header file as a precondition that all functions that may use this global are to be called in the main thread only then there is no need for mutexes to guard it and you can avoid the recursive mutex.  I think that is true from inspection of the entire PR, but not obvious.  The functions would be `ValidateAuth`, `HasRefreshToken`, `UnlinkAccount`